### PR TITLE
Simplify primitive types cast

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/scalar/cast/CastFunctionResolver.java [hunk 1]
```diff
@@ -67,4 +71,7 @@ public static FunctionInfo functionInfo(DataType dataType, DataType returnType)
         return new FunctionInfo(new FunctionIdent(functionName, ImmutableList.of(dataType)), returnType);
     }
 
+    public static String getCastFunctionName(DataType dataType) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/scalar/cast/CastFunctionResolver.java]: is this only used in tests? 
Can this also be package private? If that doesn't work I'd rather use hardcoded functionNames in the test or if you don't like that maybe add constants under a public static class Names like we do it to expose columns in the system TableInfo classes.

Comment (kovrus) [inline sql/src/main/java/io/crate/operation/scalar/cast/CastFunctionResolver.java]: it is used in two places i would probably hardcode it

**Comment by kovrus** [comment]
added two static nested classes in CastFunctionResolver, they would be necessary for array cast functions

**Comment by mfussenegger** [comment]
ack

**Pull request merged**